components/org.wso2.carbon.identity.local.auth.push.servlet/src/main/java/org/wso2/carbon/identity/local/auth/push/servlet/PushAuthServlet.java

```diff
@@ -0,0 +1,169 @@
+package org.wso2.carbon.identity.local.auth.push.servlet;
+
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.JWTParser;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONObject;
+import org.wso2.carbon.identity.local.auth.push.authenticator.PushAuthContextManager;
+import org.wso2.carbon.identity.local.auth.push.authenticator.model.PushAuthContext;
+import org.wso2.carbon.identity.local.auth.push.servlet.constant.PushServletConstants;
+import org.wso2.carbon.identity.local.auth.push.servlet.impl.PushAuthStatusCacheManagerImpl;
+import org.wso2.carbon.identity.local.auth.push.servlet.internal.PushServletDataHolder;
+import org.wso2.carbon.identity.notification.push.common.PushChallengeValidator;
+import org.wso2.carbon.identity.notification.push.common.exception.PushTokenValidationException;
+import org.wso2.carbon.identity.notification.push.device.handler.exception.PushDeviceHandlerException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.text.ParseException;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static org.wso2.carbon.identity.local.auth.push.servlet.constant.PushServletConstants.AUTH_RESPONSE;
+
+/**
+ * Servlet for handling authentication requests sent from mobile device.
+ */
+public class PushAuthServlet extends HttpServlet {
+
+    private static final long serialVersionUID = 3471640151205811758L;
+    private static final Log log = LogFactory.getLog(PushAuthServlet.class);
+    private static final PushAuthStatusCacheManager pushAuthStatusCacheManager = new PushAuthStatusCacheManagerImpl();
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+
+        handleMobileResponse(request, response);
+    }
+
+    private void handleMobileResponse(HttpServletRequest request, HttpServletResponse response)
```

Comment:
Lets use the word device instead of mobile as it was the word used in other places